src/uu/split/src/split.rs

```diff
@@ -65,6 +67,82 @@ pub fn uumain(args: impl uucore::Args) -> UResult<()> {
     }
 }
 
+/// If the same separator (with the same value) was used multiple times - `split` should NOT fail
+/// If the separator was used multiple times but with different values - `split` should fail
+/// Unfortunately Clap cannot support both conditions simultaneously, so it is setup to fail on the second condition
+/// For the first condition - this function will find if there are multiple separator options with the same value
+/// and remove duplicates, leaving only one
+/// i.e. `split -t: -t:` will become `split -t:`
+/// However it will leave other values untouched, so
+/// `split -t: -t: -t,` will become `split -t: -t,` and Clap will panic next as it should
+fn handle_multiple_separator_options(args: &[OsString]) -> Vec<OsString> {
```

Comment:
do you think you could write pure rust unit tests just for this function?

Reply:
should be doable, yes. Before I do though - isn't it already 100% covered by tests from `tests/by_util/test_split.rs` ?

Reply:
They are but harder to debug in case of regression :)

Reply:
turns out this function is not necessary - refactored using Clap `ArgAction::append` - kudos to @tertsdiepraam for suggestion :)